tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,271 @@
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83593996
+    Bug id - https://bugzilla.redhat.com/show_bug.cgi?id=2305677
+    1. Configure a cluster that have more than four OSD nodes
+    2. Select an OSD node and drain the node
+    3. Parallely execute the ceph mgr fail command
+    4. Check that exception occurs on the cluster
+    5. Perform the following workaround  steps-
+          5.1 ceph config-key rm mgr/cephadm/osd_remove_queue
+          5.2 ceph mgr fail
+    6. If exception occured then check for the Traceback logs and perform the workaround
+    7. After the verification add the drained node back to the cluster
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_obj = MgrWorkflows(node=cephadm)
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    service_obj = ServiceabilityMethods(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+    config = kw["config"]
+
+    replicated_config = config.get("replicated_pool")
+    pool_name = replicated_config["pool_name"]
+
+    if not rados_obj.create_pool(pool_name=pool_name):
+        log.error("Failed to create the  Pool")
+        return 1
+
+    rados_obj.bench_write(pool_name=pool_name, byte_size="5M", rados_write_duration=60)
+    mgr_daemon = Thread(
+        target=background_mgr_task, kwargs={"mgr_object": mgr_obj}, daemon=True
+    )
+    # Printing the hosts in cluster
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    log.info(f"The hosts in the cluster before starting the test are - {out}")
+
+    mgr_host_object_list = []
+    for node in ceph_nodes:
+        if node.role == "mgr":
+            mgr_host_object_list.append(node)
+
+    mgr_daemon_list = mgr_obj.get_mgr_daemon_list()
+
+    log_lines = [
+        "mgr load Traceback",
+        "TypeError: __init__() got an unexpected keyword argument 'original_weight'",
+    ]
+
+    ceph_version = rados_obj.run_ceph_command(cmd="ceph version")
+    log.info(f"Current version on the cluster : {ceph_version}")
+    match_str = re.match(
+        r"ceph version (\d+)\.(\d+)\.(\d+)-(\d+)", ceph_version["version"]
+    )
+    major, minor, patch, build = match_str.groups()
+    bug_exists = False
+    exceptional_flag = False
+    if int(major) < 18:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) < 2:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) < 1:
+        bug_exists = True
+    elif int(major) == 18 and int(minor) == 2 and int(patch) == 1 and int(build) <= 234:
+        bug_exists = True
+
+    # Check the labels
+    cmd_host_ls = "ceph orch host ls"
+    out = rados_obj.run_ceph_command(cmd=cmd_host_ls)
+    drain_host = None
+    for node in out:
+        if "_no_schedule" in node["labels"]:
+            drain_host = node["hostname"]
```

Comment:
break once the host with label is found.

Reply:
I added the break statement.